golang-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Initialize the module
+  - name: golang
+    args: ['go', 'mod', 'init', 'github.com/GoogleCloudPlatform/cloud-build-samples/golang']
+
+  # Clean up dependencies
+  - name: golang
+    args: ['go', 'mod', 'tidy']
+
+  # Build application
+  - name: golang
+    args: ['go', 'build', '.']
+
+  # Run tests and save to file
+  - name: golang
+    entrypoint: /bin/bash
+    args: 
+      - -c
+      - |
+        go test . -v -json > ${SHORT_SHA}_test_log.json
+
+  # Docker Build
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['build', '-t', 
+           'us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA', '.']
```

Comment:
I imagine this is a repo convention. In practice I might parameterize the region.

Reply:
Yeah we do use it elsewhere in the repo. This is one of the downsides of artifact registry -- it requires *so* many variables.  We'll probably keep this hardcoded in, but make sure to mention it in the docs.  @nityaravi WDYT?